src/loader/GLTFLoader.ts

```diff
@@ -233,10 +233,17 @@ class GLTFLoader {
         return levels;
       }, []);
 
-      texturesByLevel.forEach(async levelTextures => {
-        const texturesLoaded = await Promise.all(levelTextures.map(({ index }) => gltf.parser.getDependency("texture", index) as Promise<THREE.Texture>))
+      const loaded = texturesByLevel.map(() => false);
 
+      texturesByLevel.forEach(async (levelTextures, level) => {
         // Change textures when all texture of the level loaded
+        const texturesLoaded = await Promise.all(levelTextures.map(({ index }) => gltf.parser.getDependency("texture", index) as Promise<THREE.Texture>))
+        const higherLevelLoaded = loaded.slice(level + 1).some(val => !!val);
```

Comment:
Instead of `loaded`, you can compare it with a number such as `currentHighestLevel(?)`.

Reply:
That could be one way. I won't change it due to debug purposes though.